curated_transformers/util/pytorch.py

```diff
@@ -0,0 +1,36 @@
+from typing import Callable, List, Optional
+from dataclasses import dataclass
+from torch.nn import Module
+
+
+@dataclass
+class ModuleIterator:
+    # Current module.
+    module: Module
+    # Module name.
+    name: str
+    # Current dot path of the module. Includes the name.
+    prefix: str
+    # Parent module.
+    parent: Optional[Module]
+
+
+def apply_to_module(module: Module, func: Callable[[ModuleIterator], None]):
+    """Apply a function the module and its submodules in a breadth-first
+    fashion.
+
+    :param module:
+        Root module.
+    :param func:
+        A callable that takes a module iterator as its argument.
+    """
+    queue: List[ModuleIterator] = [ModuleIterator(module, "", "", None)]
+
+    while queue:
+        itr = queue.pop(0)
+        func(itr)
+
+        for name, child in itr.module._modules.items():
```

Comment:
Can we use `named_children` here?

https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.named_children

Reply:
Done.